### PR TITLE
Grid work

### PR DESCRIPTION
**Pull request by bluntelk**
Make the /grid endpoint super fast

fixup the grid calc and tests

### CODE REVIEW

File: lib/tile_grid/grid.go [hunk 1]
```diff
@@ -47,15 +48,30 @@ func init() {
...
 	def := GlobalDef{}
 	err := json.Unmarshal([]byte(data), &def)
 	if nil != err {
 		return err
 	}
-	worldGrid = make(map[string]GlobeIndexSpecialTile)
 	for i, tile := range def.GlobeIndexSpecialTiles {
 		worldGrid["tile"+strconv.Itoa(i)] = tile
 	}
+
```
Comment (shaneshort) [inline lib/tile_grid/grid.go:60]: does this need to stay here?

**Review by shaneshort — APPROVED** [review]

**Comment by shaneshort** [comment]
looks like we've got a test failing:
--- FAIL: Test_LookupTile (0.00s)
[146](https://github.com/plane-watch/pw-pipeline/runs/6723372720?check_suite_focus=true#step:5:147)
    --- FAIL: Test_LookupTile/Perth_is_found (0.00s)


may need to fix that up before merging

File: cmd/pw_ws_broker/web.go [hunk 2]
```diff
@@ -175,16 +199,25 @@ func (bw *PwWsBrokerWeb) indexPage(w http.ResponseWriter, r *http.Request) {
...
+	}
 	w.Header().Set("Content-Type", "application/json")
 	w.Header().Set("Cross-Origin-Resource-Policy", "cross-origin")
 	w.Header().Set("Access-Control-Allow-Origin", "*")
-	json := jsoniter.ConfigFastest
-	buf, err := json.MarshalIndent(grid, "", "  ")
-	if nil != err {
-		w.WriteHeader(500)
+	w.Header().Set("ETag", gridJsonPayloadETag)
+	w.Header().Set("Cache-Control", "public, max-age=86400")
```
Comment (tardoe) [inline cmd/pw_ws_broker/web.go:210]: We should really have a caching / nginx layer in front that controls this eventually but okay for now.

**Review by tardoe — APPROVED** [review]
LGTM

File: lib/tile_grid/grid.go (hunk 1 above)
Comment (bluntelk) [reply lib/tile_grid/grid.go:60]: i just moved the worldGrid init up a few lines so I can easily comment/uncomment the 2 blocks

**Pull request merged**